### DISCUSSION TITLE
ObservableProperty not working for me

### DISCUSSION THREAD

**Discussion by humbertoadan** [CommunityToolkit/dotnet#357] [Q&A] [answered]
Hello,

I'm new to the CommunitToolkit, so I'm just wondering if I'm doing something wrong or is it a bug. I'm using the lastest version 8.0 preview4. This is a part of my ViewModel code:

```
public partial class CategoriesViewModel : ObservableObject
{
    private readonly IRepository<Category> _categoriesRepository;
    readonly ObservableCollection<Category> observableCategories;

    public CategoriesViewModel()
    {
        _categoriesRepository = App.GetService<IRepository<Category>>();
        observableCategories = new ObservableCollection<Category>();
    }

    [ObservableProperty]
    [NotifyPropertyChangedFor(nameof(CategorySearched))]
    private string _categorySearch;

   public string CategorySearched=> $"Category Searched:{CategorySearch}";
```

And this is part of my View xaml code:
```
<Page
    x:Class="gizmo_pos_main.Views.CategoriesPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:Interactions="using:Microsoft.Xaml.Interactions.Core"
    xmlns:Interactivity="using:Microsoft.Xaml.Interactivity"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:local="using:gizmo_pos_main.ViewModels"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d">

    <Page.DataContext>
        <local:CategoriesViewModel />
    </Page.DataContext>

<StackPanel
            Grid.Column="1"
            BorderThickness="2"
            Orientation="Vertical">
            <TextBox
                x:Name="CategoryTextSearch"
                Width="200"
                Margin="8"
                HorizontalAlignment="Left"
                Header="Buscar"
                Text="{Binding CategorySearch}">
            </TextBox>
            <TextBlock Text="{Binding CategorySearched}" />
```
Right now, I'm trying to test displaying on a TextBlock the text the is entered on the TextBox and update the TextBlock with the correspoding input. Right now, it doesn't work. 

I'm trying to test this library ever since I saw this video https://www.youtube.com/watch?v=aCxl0z04BN8&t=838s from James Montemagno doing the same thing. He uses an older package version but the concept is the same. I'm trying to replicate the same in my project and it just doesn't work. Any help is appreciated. Thanks.

**Comment by Sergio0694** [comment, marked as answer]
I assume this is UWP/WinUI 3? You seem to be missing a `Mode=TwoWay` in your `{Binding CategorySearch}` 🙂

**Reply by humbertoadan** [reply]
Thank you very much! The text is now displayed in the TextBlock, although it only displays it after the TextBox loses focus. When I watched the video, the text block updated as he typed in the TextBox. But perhaps that has to do more about something missing from my XAML code. Or maybe there is a difference since he used MAUI in the video and I'm using WinUI 3. Anyway, thank you for helping me. Really appreciate it. 👍

**Reply by Sergio0694** [reply]
> "although it only displays it after the TextBox loses focus"

Likely just a different default on MAUI.
Add `UpdateSourceTrigger=PropertyChanged` to your binding as well, should do the trick 🙂

**Reply by humbertoadan** [reply]
That did it! Thank you! 🙂👍
